scripts/md-gen/controller-config/main.go

```diff
@@ -0,0 +1,472 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package main
+
+import (
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"os"
+	"path/filepath"
+	"reflect"
+	"sort"
+	"strings"
+
+	"github.com/juju/collections/set"
+	"github.com/juju/schema"
+	"gopkg.in/yaml.v3"
+
+	"github.com/juju/juju/controller"
+	"github.com/juju/juju/testing"
+)
+
+// bidirectional mapping between key and constant name
+// e.g. "agent-ratelimit-max" <--> "AgentRateLimitMax"
+// These are filled by iterating through the AST of config.go
+var (
+	keyForConstantName = map[string]string{}
+	constantNameForKey = map[string]string{}
+)
+
+// Generate Markdown documentation based on the contents of the
+// github.com/juju/juju/controller package.
+func main() {
+	outputDir := os.Getenv("DOCS_DIR")              // directory to write output to
+	packagePath := os.Getenv("CONTROLLER_PKG_PATH") // path to local github.com/juju/juju/controller package
+
+	data := map[string]*keyInfo{}
+
+	// Gather information from various places
+	fillFromAST(data, packagePath)
+	fillFromConfigType(data)
+	fillFromAllowedUpdateConfigAttributes(data)
+	fillFromNewConfig(data)
+
+	_ = os.MkdirAll(outputDir, 0755)
```

Comment:
`check(err)`?

Reply:
if this doesn't work, no need to bring down the whole enterprise IMO. This is just here to prevent failures - it shouldn't cause them.